akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/BehaviorTestKitImpl.scala

```diff
@@ -132,7 +133,11 @@ private[akka] final class BehaviorTestKitImpl[T](_path: ActorPath, _initialBehav
     try {
       context.setCurrentActorThread()
       try {
-        currentUncanonical = Behavior.interpretMessage(current, context, message)
+        //we need this to handle message adapters related messages
+        val intercepted = BehaviorTestKitImpl.Interceptor.inteceptBehaviour(current, context)
+        currentUncanonical = Behavior.interpretMessage(intercepted, context, message)
+        //notice we pass current and not intercepted, this way Behaviors.same will be resolved to current which will be intercepted again on the next message
+        //otherwise we would have risked intercepting and already intercepted behaviour (or explicitly checking if the current behaviour is already intercepted by us)
```

Comment:
Good to clarify this with a comment, some minor spelling suggestion:
```suggestion
        //otherwise we would have risked intercepting an already intercepted behavior (or would have had to explicitly check if the current behavior is already intercepted by us)
```